### DISCUSSION TITLE
The duplicates method does not work correctly

### DISCUSSION THREAD

**Discussion by deraak** [laravel/framework#42309] [Q&A]
- Laravel Version: 9.7.0
- PHP Version: 8.1.2
- Database Driver & Version:

### Description:
The issue:

`
collect(['A', 'A', 'B', 'B', 'C', 'C', 'C', 'C', 'D'])->duplicates()->all(); // it returns ['A' , 'B' , 'C', 'C', 'C']
`
also the duplicatesStrict has this issue too

**Comment by sjokkateer** [comment]
Could you please also add what you expect the method to return?

The method's documentation [Collection::class#L289](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Collections/Collection.php#L289) mentions it retrieving the duplicate items in the collection, which it seems to do in your provided example. It is accounting for one of the items being the original and hence not a duplicate, thus not included.

**Comment by crissi** [comment]
I think the behaviour is expected. If you want unique duplicates you can do:
`collect(['A', 'A', 'B', 'B', 'C', 'C', 'C', 'C', 'D'])->duplicates()->unique()->all();`

**Reply by deraak** [reply]
In this example, the duplicates method return items which are duplicate and I think this behaviour  is incorrect

**Reply by rodrigopedra** [reply]
It is the intended behavior, as they are returned with their key.

Mind this method returns **the duplicate values in a list**, not **the values that have duplicates in a list**.

So let's say a user uploads a list of data and you want to highlight the duplicate ones. You can use this:

~~~php
$keys = collect(['A', 'A', 'B', 'B', 'C', 'C', 'C', 'C', 'D'])->duplicates()->keys();
~~~

To get the keys from the duplicate values, pass it to the frontend and highlight all duplicate values, regardless if it is a second, third or any duplicate of any previous values.

Maybe to your use-case you'd like it to behave differently, but there is certainly a valid use-case for current behavior.

If you want all the values that have duplicates, you can use @crissi suggestion and call `->duplicates()` and then `->unique()` to remove any repeated duplicates.
